src/platform/packages/shared/response-ops/form-generator/src/form.stories.tsx

```diff
@@ -0,0 +1,236 @@
+  render: () => {
+    return <Form connectorSchema={GreyNoiseConnectorSchema} onSubmit={submit} />;
+  },
+};
+
+export const ShodanConnector: StoryObj = {
+  render: () => {
+    return <Form connectorSchema={ShodanConnectorSchema} onSubmit={submit} />;
+  },
+};
+
+export const UrlVoidConnector: StoryObj = {
+  render: () => {
+    return <Form connectorSchema={UrlVoidConnectorSchema} onSubmit={submit} />;
+  },
+};
+
+export const VirusTotalConnector: StoryObj = {
+  render: () => {
+    return <Form connectorSchema={VirusTotalConnectorSchema} onSubmit={submit} />;
+  },
+};
+
+// this is not the object that would be created as SingleConnectorFile, this is the internal
+// representation after parsing a SingleConnectorFile
+const webhookConnectorFormSchema = z.object({
+  name: z.string().min(1, { message: 'Name cannot be empty' }).meta({
+    widget: 'text',
+    label: 'Connector Name',
+  }),
+  method: z.enum(['POST', 'PUT', 'GET', 'DELETE']).meta({
+    widget: 'select',
+    label: 'Method',
+    default: 'POST',
+  }),
+  url: z.url().meta({ widget: 'text', label: 'URL' }),
+  authType: z
+    .discriminatedUnion('type', [
+      z.object({ type: z.literal('none') }).meta({
+        label: 'None',
+      }),
+      z
+        .object({
+          type: z.literal('basic'),
+          username: z.string().min(1, { message: 'Username cannot be empty' }).meta({
+            widget: 'text',
+            label: 'Username',
+          }),
+          password: z.string().min(1, { message: 'Password cannot be empty' }).meta({
+            widget: 'password',
+            label: 'Password',
+          }),
+        })
+        .meta({ label: 'Basic Authentication' }),
+      z
+        .object({
+          type: z.literal('bearer'),
+          token: z.string().min(1, { message: 'Token cannot be empty' }).meta({
+            widget: 'password',
+            label: 'Token',
+          }),
+        })
+        .meta({ label: 'Bearer Token' }),
+      z
+        .object({
+          type: z.literal('headers'),
+          headers: z.record(z.string(), z.string()).meta({
+            widget: 'keyValue',
+            label: 'Headers',
+          }),
+        })
+        .meta({
+          label: 'Headers',
+        }),
+    ])
+    .meta({
+      widget: 'formFieldset',
+      label: 'Authentication',
+      default: 'basic',
+    }),
+});
+
+const abuseIPDBConnectorSchema = z.object({
+  authType: z
+    .discriminatedUnion('type', [
+      z
+        .object({
+          type: z.literal('apiKey'), // this literal is irrelevant
+          Key: z.string().min(1, { message: 'API Key cannot be empty' }).meta({
+            widget: 'password',
+            label: 'API Key',
+            placeholder: 'Your AbuseIPDB API Key',
+          }),
+        })
+        .meta({
+          label: 'Headers',
+        }),
+    ])
+    .meta({ widget: 'formFieldset', label: 'Authentication' }),
+});
+
+const alienVaultOTXConnectorSchema = z.object({
```

Comment:
The current conversion function from [my PR](https://github.com/elastic/kibana/pull/242776/files#diff-d58418e6c9ebdb2f95bee6e8f621a0500f3dc88b0dcce0dfd94dbea1f016ddfb)

will end up with a schema like

```
z.discriminatedUnion('authType', [
  z.object({
    authType: z.literal('header'),
    headers: z.object({
      'X-OTX-API-KEY': z.string().min(1, { message: 'API Key cannot be empty' }).meta({
            widget: 'password',
            label: 'API Key',
          })
    })
}) 
```

does that work?